### PR TITLE
Adding classes for port creation

### PR DESCRIPTION
**Pull request by nknaian**
- created file to hold port classes
- used classes to create port objects and put in some basic sending code to test it out
- added comments as a template for next steps

### CODE REVIEW

File: server_ui/app/appServer.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
 const express = require('express');
-const osc = require('osc');
-const dgram = require('dgram');
 const io = require('socket.io')();
-
-
+const oscUdpPort = require('./serverDependencies/ports').oscUdpPort;
+const dgramUdpPort = require('./serverDependencies/ports').dgramUdpPort;
```
Comment (vsuha) [inline server_ui/app/appServer.js]: wow so smart

File: server_ui/app/serverDependencies/ports.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    sendData(data) {
+      this.send({
+          address: this.address,
+          args: data
+      }, this.options.remoteAddress, this.options.remotePort);
+      console.log(`Data sent over ${this.address} port: ${data}`);
+    };
+}
+
+class dgramUdpPort extends dgram.Socket {
```
Comment (mlang251) [inline server_ui/app/serverDependencies/ports.js]: class udpSocket = {
    constructor(port, callback) {
        this.socket = dgram.createSocket('udp4');
        this.socket.bind(port);
    }
}

File: server_ui/app/serverDependencies/ports.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const osc = require('osc');
+const dgram = require('dgram');
+
+var exports = {};
+
+class oscUdpPort extends osc.UDPPort {
+    constructor(port, address) {
+      super({
+          localPort: port + 1,
```
Comment (mlang251) [inline server_ui/app/serverDependencies/ports.js]: localPort: null (or undefined)

File: server_ui/app/appServer.js [hunk 4]
```diff
@@ -15,45 +13,50 @@ let server = app.listen(3000, () => {
...
+const leapToServerChannel = {
+    portCoord: new dgramUdpPort(8000),
+    currentCoord: [0, 0, 0]
+}
+leapToServerChannel.portCoord.on("message", (msg, rinfo) => {
     msg = msg.toString();
     console.log(`received message from leap: ${msg}`);
-    //TO DO: save the data and pass to the parameters
-});
+    leapToServerChannel.currentCoord = msg.split(" ");
```
Comment (mlang251) [inline server_ui/app/appServer.js]: convert array of strings into array of ints with Number('x')

**Pull request merged**